enable_disable_persistence.md

```diff
@@ -0,0 +1,180 @@
+
+The state is stored the browser's LocalStore, and is tied to the user's browser.
+
+### Manual Trigger
+
+Today, when you switch a resource between manual trigger mode and auto trigger
+mode, that state persists until you reload the Tiltfile. When the Tiltfile
+loads/reloads, the trigger mode reverts to what's specified in the Tiltfile.
+
+### systemd
+
+[systemd and
+systemctl](https://www.commandlinux.com/man-page/man1/systemctl.1.html) is
+probably the most widely known service manager. You tend to see its verbs in
+other service management tools (like
+[Edward](http://engblog.yext.com/edward/commands/#start)).
+
+`systemd` is a persistent daemon that runs admin-level services on linux.
+
+You use `systemctl start` and `systemctl stop` to start/stop services.
+
+And `systemctl enable` and `systemctl disable` to enable/disable services to
+auto-boot at system startup.
+
+Note that `enable` does not start a service immediately. In other words, systemd
+models "start on boot" and "start now" as independent actions.
+
+## Product Specification
+
+### Run all, Disable from UI
+
+1. Write a Tiltfile with services Paulie and Roberta
+2. Run `tilt up` -> Paulie and Roberta are enabled.
+4. Click "disable" on "Paulie" -> Roberta is enabled, Paulie is disabled
+5. Ctrl-c and re-run `tilt up` -> Roberta is enabled, Paulie is disabled
+
+Possible Principle:
+
+`tilt up` (without arguments) always inherits all the enable/disable overides
+from the previous Tilt session.
+
+### Specify services, Disable from UI
+
+1. Write a Tiltfile with services Paulie and Roberta and Joe
+2. Run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled.
+4. Click "disable" on "Paulie" -> Roberta is enabled, Joe and Paulie are disabled.
+5. Ctrl-c and re-run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled.
+
+Possible Principle:
+
+Tilt always respects arguments to `tilt up` when it starts.
+
+### Specify services, Enable from UI
+
+1. Write a Tiltfile with services Paulie and Roberta and Joe
+2. Run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled.
+4. Click "enable" on "Joe" -> Paulie and Roberta and Joe are enabled.
+5. Ctrl-c and re-run `tilt up paulie roberta` -> Paulie and Roberta are enabled, Joe is disabled. 
+
+Possible Principle:
+
+Tilt always respects arguments to `tilt up` when it starts.
+
+Your UI change has been wiped out by the `tilt up` args.
+
+### Run all, Disable from UI, Change Args
+
+1. Write a Tiltfile with services Paulie and Roberta
+2. Run `tilt up` -> Paulie and Roberta are enabled.
+4. Click "disable" on "Paulie" -> Roberta is enabled, Paulie is disabled.
+5. Run `tilt args paulie roberta` in a separate terminal. -> Tiltfile re-executes, Paulie and Roberta are enabled. 
+
+Possible Principle:
+
+Changing the tilt args wipes out any enabled/disabled services from the UI.
+
+### Run one service, Enable from UI, Change Args
```

Comment:
I agree with all the other specs but this one feels too much like an exception. I'd prefer to `tilt args` to do the same things every time. (So in this case running `tilt args paulie` the second time would make it so that only `paulie` is enabled.

Reply:
Ya. This one comes up A LOT in kubernetes-land. 

The classic example is `kubectl apply` - if the input doesn't change, it does nothing on the second apply, but lots of people expect it to restart pods.

we probably could make this work but it'd be messy.